core/services/relay/ethereum/ethereum.go

```diff
@@ -0,0 +1,211 @@
+}
+
+// No peristent subservices to close on relay close
+func (r *Relayer) Close() error {
+	return nil
+}
+
+// Always ready
+func (r *Relayer) Ready() error {
+	return nil
+}
+
+// Always healthy
+func (r *Relayer) Healthy() error {
+	return nil
+}
+
+func (r *Relayer) NewOCR2Provider(externalJobID uuid.UUID, s interface{}) (relay.OCR2Provider, error) {
+	spec, ok := s.(OCR2Spec)
+	if !ok {
+		return nil, errors.New("unsuccessful cast to 'ethereum.OCR2Spec'")
+	}
+	chain, err := r.chainSet.Get(spec.ChainID)
+	if err != nil {
+		return nil, err
+	}
+	// TODO: more validation
+	contractAddress := common.HexToAddress(spec.ContractID.String)
+
+	contract, err := offchain_aggregator_wrapper.NewOffchainAggregator(contractAddress, chain.Client())
+	if err != nil {
+		return nil, errors.Wrap(err, "could not instantiate NewOffchainAggregator")
+	}
+
+	contractFilterer, err := ocr2aggregator.NewOCR2AggregatorFilterer(contractAddress, chain.Client())
+	if err != nil {
+		return nil, errors.Wrap(err, "could not instantiate NewOffchainAggregatorFilterer")
+	}
+
+	contractCaller, err := ocr2aggregator.NewOCR2AggregatorCaller(contractAddress, chain.Client())
+	if err != nil {
+		return nil, errors.Wrap(err, "could not instantiate NewOffchainAggregatorCaller")
+	}
+
+	ocrDB := ocr2.NewDB(r.db.DB, spec.ID, r.lggr)
+
+	tracker := ocr2.NewOCRContractTracker(
+		contract,
+		contractFilterer,
+		contractCaller,
+		chain.Client(),
+		chain.LogBroadcaster(),
+		spec.ID,
+		r.lggr,
+		r.db,
+		ocrDB,
+		chain.Config(),
+		chain.HeadBroadcaster(),
+	)
+
+	offchainConfigDigester := evmutil.EVMOffchainConfigDigester{
+		ChainID:         chain.Config().ChainID().Uint64(),
+		ContractAddress: contractAddress,
+	}
+
+	if spec.IsBootstrap {
+		// Return early if bootstrap node (doesn't require the full OCR2 provider)
+		return &ocr2Provider{
+			tracker:                tracker,
+			offchainConfigDigester: offchainConfigDigester,
+		}, nil
+	}
+
+	reportCodec := evmreportcodec.ReportCodec{}
+
+	contractABI, err := abi.JSON(strings.NewReader(ocr2aggregator.OCR2AggregatorABI))
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get contract ABI JSON")
+	}
+
+	if !spec.TransmitterID.Valid {
+		return nil, errors.New("transmitter address is required")
+	}
+
+	transmitterAddress := common.HexToAddress(spec.TransmitterID.String)
+	strategy := txm.NewQueueingTxStrategy(externalJobID, chain.Config().OCRDefaultTransactionQueueDepth(), false)
+
+	contractTransmitter := ocr2.NewOCRContractTransmitter(
+		contract.Address(),
+		contractCaller,
+		contractABI,
+		ocrcommon.NewTransmitter(chain.TxManager(), transmitterAddress, chain.Config().EvmGasLimitDefault(), strategy),
+		tracker,
+		r.lggr,
+	)
+
+	return &ocr2Provider{
+		tracker:                tracker,
+		offchainConfigDigester: offchainConfigDigester,
+		reportCodec:            reportCodec,
+		contractTransmitter:    contractTransmitter,
+	}, nil
+}
+
+type RelayConfig struct {
+	ChainID utils.Big `json:"chainID"`
```

Comment:
This ought to be a `*utils.Big` because its nullable. In the case where its missing, we fallback to `chainset.Default`

Reply:
(Otherwise it will read missing values as `0` which is almost certainly not what we want, although it might accidentally pass tests)